src/NServiceBus.Core/Routing/FileBasedDynamicRouting/FileRoutingTableFeature.cs

```diff
@@ -11,19 +12,31 @@ public FileRoutingTableFeature()
             {
                 s.SetDefault(CheckIntervalSettingsKey, TimeSpan.FromSeconds(30));
                 s.SetDefault(MaxLoadAttemptsSettingsKey, 10);
+                s.SetDefault(FilePathSettingsKey, "instance-mapping.xml");
             });
             DependsOn<RoutingFeature>();
         }
 
         protected internal override void Setup(FeatureConfigurationContext context)
         {
             var filePath = context.Settings.Get<string>(FilePathSettingsKey);
+            if (!Path.IsPathRooted(filePath))
+            {
+                filePath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, filePath);
+            }
+
+            if (!File.Exists(filePath))
```

Comment:
the main idea behind this was to avoid expensive operations when it's highly unlikely they are needed. Using the mapping file means:
* re-read & parse the file every x seconds
* add a dynamic routing rule which is invoked on every send

so the idea was to basically "disarm" the feature if no file was found which means if you plan to roll out without a file and add the mapping later (I think the only valid scenario would be moving a receiver endpoint from the same machine to a different machine at runtime) you need to add at least an empty file. I don't have a strong opinion on this one but I thought this may be better than setting up the file in every case.